lib/view_model/exchange/exchange_view_model.dart

```diff
@@ -776,16 +782,70 @@ abstract class ExchangeViewModelBase extends WalletChangeListenerViewModel with
 
   int get receiveMaxDigits => receiveCurrency.decimals;
 
-  bool isCanCreateTrade(Trade trade) {
+  Future<CreateTradeResult> isCanCreateTrade(Trade trade) async {
     if (trade.provider == ExchangeProviderDescription.thorChain) {
       final payoutAddress = trade.payoutAddress ?? '';
       final fromWalletAddress = trade.fromWalletAddress ?? '';
       final tapRootPattern = RegExp(P2trAddress.regex.pattern);
 
       if (tapRootPattern.hasMatch(payoutAddress) || tapRootPattern.hasMatch(fromWalletAddress)) {
-        return false;
+        return CreateTradeResult(
+          result: false,
+          errorMessage: S.current.thorchain_taproot_address_not_supported,
+        );
+      }
+
+      final currenciesToCheckPattern = RegExp('0x[0-9a-zA-Z]');
+      final inputAddress = trade.inputAddress ?? '';
+      if (currenciesToCheckPattern.hasMatch(inputAddress)) {
+        final isEOAAddress = await _isExternallyOwnedAccountAddress(inputAddress);
+        print('Is EOA Address: $isEOAAddress');
+        return CreateTradeResult(
+          result: isEOAAddress,
+          errorMessage: !isEOAAddress ? S.current.thorchain_contract_address_not_supported : null,
+        );
       }
     }
-    return true;
+    return CreateTradeResult(result: true);
+  }
+
+  Future<bool> _isExternallyOwnedAccountAddress(String receivingAddress) async {
+    final isEOAAddress = !(await _isContractAddress('eth', receivingAddress));
```

Comment:
is hardcoding `eth` here right? it might be polygon (matic) or other L2 chains when we integrate them, why not just use the `to` cryptoCurrency's tag, which should be correct already and we don't need to hardcode it